pxteditor/events.ts

```diff
@@ -0,0 +1,46 @@
+namespace pxt.events {
+
+    export type EditorType = 'blocks' | 'js';
+
+    export interface Event {
+        type: string;
+        editor: EditorType;
+    }
+
+    export abstract class AbstractEvent implements Event {
```

Comment:
do we need classes, we can typically define these messages as bare interface

Reply:
Switching to interfaces